### DISCUSSION TITLE
Xformers or sdp attention support?

### DISCUSSION THREAD

**Discussion by just-someguy** [KoboldAI/KoboldAI-Client#310] [Ideas]
I can barely find any mention on the internet at all about KoboldAI using an improved attention model like xformers or sdp_attention, with just a few people on reddit saying they wish it were a feature. These greatly improve token generation, and I'd like to see them added. Is this something being worked on? Or not necessarily a feature being pursued?

**Comment by henk717** [comment]
Currently not being persued since we have other priorities. One of which is a backend overhaul that allows the community to add stuff to the backend much more easily. We are currently mostly in a phase where we are aiming to get things stable so we can ship a new main version of KoboldAI and since its by far the biggest update we ever worked on thats taking a while.

But once the new backend overhaul is in people from the community might be able to add this before we have time for these kinds of additions again.
